src/OpenSleigh.Transport.RabbitMQ/RabbitPublisher.cs

```diff
@@ -30,37 +30,40 @@ public Task PublishAsync(IMessage message, CancellationToken cancellationToken =
             if (message is null)
                 throw new ArgumentNullException(nameof(message));
 
-            var context = _publisherChannelFactory.Create(message);
-
-            var encodedMessage = _encoder.Encode(message);
-
-            var properties = context.Channel.CreateBasicProperties();
-            properties.Persistent = true;
-            properties.Headers = new Dictionary<string, object>()
+            return Task.Run(() =>
```

Comment:
Why you are doing this?

This is basically always a code smell to me, in a framework I would also say that is a worst practice. In a high-load scenario you would end up starving the thread pool with threads that is waiting on external resources (BasicPublish) or just sleeping (while executing the retry for polly policy).

Another big problem is that the [IModel (aka Channel) is not thread safe](https://www.rabbitmq.com/dotnet-api-guide.html#concurrency-channel-sharing) and must not be shared by threads that publish on them.

To achieve this level of asynchronously, you can use a producer-consumer pattern, and implement a dispatcher, [we had this for a long time in EasyNetQ](https://github.com/EasyNetQ/EasyNetQ/blob/47b17c9471474a29f2103c0bc7fb9f8d78ff2726/Source/EasyNetQ/Producer/ClientCommandDispatcherSingleton.cs#L10), but then we switched to a [`Semaphore` implementation](https://github.com/EasyNetQ/EasyNetQ/blob/3cc90d1519160d97ad1e9240a946c824db92e280/Source/EasyNetQ/Producer/PersistentChannel.cs#L56).

Reply:
that was not meant to be there, I was just testing something and forgot to remove it. Good catch!